src/ddmd/expressionsem.d

```diff
@@ -7619,6 +7619,24 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         {
             if (t2.ty == Tpointer)
             {
+                // https://issues.dlang.org/show_bug.cgi?id=11006
+                Type p1 = t1.nextOf();
+                Type p2 = t2.nextOf();
+
+                if (!p1.equivalent(p2))
+                {
+                    // Types don't match unless the have the same size
+                    if (p1.size() != p2.size())
```

Comment:
According to spec, subtracting pointers pointing to objects of different types (regardless of size) is an error. The result obviously would be the same if you cast one of the pointers to the same type as the other one, but may be actually be a mistake. What are your opinions on being more strict, as the spec dictates?

Reply:
Yeah, I'm for following the spec.  I'll update the PR.

Reply:
My understanding is the `Type.equivalent()` function is all that's needed.  I think the size comparison in the previous commit was unnecessary. 

Again, I think this is ready to go unless I'm missing something.